pkg/api/norman/customization/cred/store.go

```diff
@@ -77,9 +80,16 @@ func Validator(request *types.APIContext, schema *types.Schema, data map[string]
 type Store struct {
 	types.Store
 	NodeTemplateLister v3.NodeTemplateLister
+	ProvClusterCache   provv1.ClusterCache
 }
 
 func (s *Store) Delete(apiContext *types.APIContext, schema *types.Schema, id string) (map[string]interface{}, error) {
+	if provClusters, err := s.ProvClusterCache.GetByIndex(cluster.ByCloudCred, id); err != nil {
+		return nil, err
+	} else if len(provClusters) > 0 {
+		return nil, httperror.NewAPIError(httperror.InvalidAction, fmt.Sprintf("cloud credential is currently referenced by provisioning cluster %s", provClusters[0].Name))
```

Comment:
```suggestion
		return nil, httperror.NewAPIError(httperror.InvalidAction, fmt.Sprintf("cloud credential is currently referenced by at least one provisioning cluster %s", provClusters[0].Name))
```

Reply:
The message here is consistent with the node template statement below. I would prefer to keep it as is, for consistency sake.

Reply:
I did not realize that. 
It sounds good to me then.